src/Skinner.js

```diff
@@ -42,7 +48,25 @@ class Skinner {
                 "If you want to change the active skin, call resetSkin first"
             );
         }
-        this.components = skinObject;
+        this.components = {};
+        var compKeys = Object.keys(skinObject.components);
+        for (var i = 0; i < compKeys.length; ++i) {
+            var comp = skinObject.components[compKeys[i]];
+            this.addComponent(compKeys[i], comp);
+        }
+    }
+
+    addComponent(name, comp) {
+        var slot = name;
+        if (comp.replaces !== undefined) {
```

Comment:
This would be another option, although I like the information that RoomDNDView is a replacement for RoomTile being in RoomDNDView.js itself and not having to refer to a separate mapping file. A third option would be to not default the 'slot' to the name of the component, ie. force all components to specify a 'replaces' (and probably rename it) which would make it more explicit.

I really don't mind on the special-casing of last path segments: I just wrote it so you could say either replaces: views.foo.bar or replaces: bar and it would do The Right Thing. If people have objections to this we can just force you to specify the full name.